crowbar_framework/app/models/service_object.rb

```diff
@@ -1852,4 +1852,25 @@ def node_changed_roles?(node, old_role, new_role)
     roles_in_new = new_role.elements.keys.select { |r| new_role.elements[r].include?(node) }.sort
     roles_in_old != roles_in_new
   end
+
+  # return true if no other attribute than the ignored has changed. And that the node has no other
+  # role than the roles in only_for_roles
+  def relevant_attributes_changed_if_roles?(node, old_role, new_role, ignore_attributes, only_for_roles)
+    # sort roles of only_for_roles
+    only_for_roles = only_for_roles.sort
```

Comment:
I think you can do `only_for_roles.sort!` and it will sort them in place, with no need to reassigning